tests/cpp/multidevice.cpp

```diff
@@ -16,32 +16,52 @@
 
 namespace nvfuser {
 
-void MultiDeviceTest::SetUp() {
-  NVFuserTest::SetUp();
-
+MultiDeviceTest::MultiDeviceTest() {
   communicator = getOrCreateCommunicator();
   tensor_options =
       at::TensorOptions().dtype(at::kFloat).device(communicator->device());
-  if (getNvFuserEnv("MULTIDEVICE_DEBUG_PRINT")) {
-    debug_print = true;
-  }
-  if (getNvFuserEnv("MULTIDEVICE_DEBUG_BARRIER") &&
-      communicator->is_available()) {
-    do_barrier_at_test = true;
-  }
-  if (getNvFuserEnv("MULTIDEVICE_DISABLE_SKIP")) {
-    disable_skip = true;
+  debug_print = getNvFuserEnv("MULTIDEVICE_DEBUG_PRINT") != nullptr;
+  do_barrier_at_test =
+      (getNvFuserEnv("MULTIDEVICE_DEBUG_BARRIER") != nullptr &&
+       communicator->is_available());
+  disable_skip = getNvFuserEnv("MULTIDEVICE_DISABLE_SKIP") != nullptr;
+}
+
+MultiDeviceTest::~MultiDeviceTest() {
+  if (do_barrier_at_test && communicator->is_available()) {
```

Comment:
I think we want to keep the optional barrier in TearDown because after each test we want to ensure all devices completed before moving onto the next test. Then, in the destructor, we want a non-optional barrier. 
wdyt?

Reply:
https://google.github.io/googletest/faq.html#CtorVsSetUp

> For each TEST_F, GoogleTest will create a fresh test fixture object, immediately call SetUp(), run the test body, call TearDown(), and then delete the test fixture object.

I know it's a bit counter intuitive but the destructor is actually called around the same time as TearDown.

Reply:
Got it! Thanks for the explanation!